flutter_appauth/ios/Classes/FlutterAppauthPlugin.m

```diff
@@ -167,18 +170,34 @@ - (void)performAuthorization:(OIDServiceConfiguration *)serviceConfiguration cli
     UIViewController *rootViewController =
     [UIApplication sharedApplication].delegate.window.rootViewController;
     if(exchangeCode) {
-        _currentAuthorizationFlow = [OIDAuthState authStateByPresentingAuthorizationRequest:request presentingViewController: rootViewController
-                                                                                   callback:^(OIDAuthState *_Nullable authState,
-                                                                                              NSError *_Nullable error) {
-                                                                                       if(authState) {
-                                                                                           result([self processResponses:authState.lastTokenResponse authResponse:authState.lastAuthorizationResponse]);
-                                                                                           
-                                                                                       } else {
-                                                                                           [self finishWithError:AUTHORIZE_AND_EXCHANGE_CODE_ERROR_CODE message:[self formatMessageWithError:AUTHORIZE_ERROR_MESSAGE_FORMAT error:error] result:result];
-                                                                                       }
-                                                                                   }];
+
+        NSObject<OIDExternalUserAgent> *agent = NULL;
+        if (preferEphemeral) {
+            agent = [[OIDExternalUserAgentIOSNoSSO alloc]
+                initWithPresentingViewController:rootViewController];
+        } else {
+            agent = [[OIDExternalUserAgentIOS alloc]
+                initWithPresentingViewController:rootViewController];
+        }
```

Comment:
It's only been repeated twice but would you be able to refactor this block of code out to a method that returns the appropriate agent?